mojap_metadata/converters/glue_converter/__init__.py

```diff
@@ -526,9 +600,127 @@ def generate_to_meta(self, database: str, table: str) -> Metadata:
+                "glue_table_properties to the table properties in Glue Catalog."
+            )
+        else:
+            table_parameters[key] = value
+
+    return table_input
+
+
+def _get_table_parameters(
+    metadata: Metadata,
+    table_parameters_resp: dict,
+    glue_table_properties: List[str] = None,
+    update_primary_key: bool = False,
+) -> dict:
+    """
+    Adds glue_table_properties and/or primary_key from the table properties in
+    the Glue Data Catalog to the metadata if glue_table_properties are defined
+    or update_primary_key is True. Set glue_table_properties to "*" to
+    add glue_table_properties to the metadata with all available table properties
+    from the Glue Data Catalog.
+
+    Args:
+        metadata (Metadata): Metadata object.
+        table_parameters_resp (dict): A dictionary defining the table properties in the
+        Glue Data Catalog.
+        glue_table_properties (List[str], optional): List of table properties to get
+        from Glue Data Catalog if they exist. Set to "*" to retrieve all table
+        properties. Defaults to None.
+        update_primary_key (bool, optional): Set to True to update the primary_key value
+        in the metadata with the primary_key table property from Glue Data Catalog if it
+        exists. Defaults to False.
+
+    Raises:
+        TypeError: If primary key value is not of type list in the Glue Data Catalog.
+        ValueError: If primary_key cannot be evaluated by ast.literal_eval.
+
+    Returns:
+        dict: Updated metadata dictionary with glue table properties and primary key
+        from the Glue Data Catalog.
+    """
+    metadata_dict = metadata.to_dict()
+    table_properties_keys = table_parameters_resp.keys()
+
+    if "primary_key" in table_properties_keys and update_primary_key:
+        try:
+            value = ast.literal_eval(table_parameters_resp["primary_key"])
+            if not isinstance(value, list):
+                raise TypeError(
+                    "ast.literal_eval cannot evaluate the primary_key value "
+                    "  type to list. Please check Glue Data Catalog"
+                )
+        except (ValueError, SyntaxError):
+            raise ValueError(
+                "ast.literal_eval cannot evaluate the primary_key value. The "
+                "primary_key value must be a Python literal structure: list. "
+                "Please check Glue Data Catalog."
+            )
+        else:
+            metadata_dict["primary_key"] = value
+
+    if glue_table_properties == "*":
+        metadata_dict["glue_table_properties"] = table_parameters_resp
+    elif glue_table_properties:
+        glue_table_properties_dict = {}
+        for property in glue_table_properties:
+            if property in table_properties_keys:
+                glue_table_properties_dict[property] = table_parameters_resp[property]
+            else:
+                warnings.warn(
```

Comment:
same as above re. including table-name in the warning.

Reply:
As above, I've included the table name in the warning and errors now.